bundler/lib/bundler/rubygems_ext.rb

```diff
@@ -275,14 +277,21 @@ def match_spec?(spec)
       def match_gem?(platform, gem_name)
         match_platforms?(platform, Gem.platforms)
       end
+    end
+
+    match_platforms_defined = Gem::Platform.respond_to?(:match_platforms?, true)
+
+    if !match_platforms_defined || Gem::Platform.send(:match_platforms?, Gem::Platform::X64_LINUX_MUSL, [Gem::Platform::X64_LINUX])
 
       private
 
+      remove_method :match_platforms? if match_platforms_defined
+
       def match_platforms?(platform, platforms)
         platforms.any? do |local_platform|
           platform.nil? ||
             local_platform == platform ||
-            (local_platform != Gem::Platform::RUBY && local_platform =~ platform)
+            (local_platform != Gem::Platform::RUBY && platform =~ local_platform)
```

Comment:
I also had to backport this bug fix from RubyGems. It's confusing (it was already like that before must support was added), but platform matching is not commutative at the moment and for things to work properly the local platform needs to be the parameter, not the receiver. This is documented on the RubyGems side, but we should improve on this anyways. Food for the future though.